blueprints/codes/__init__.py

```diff
@@ -0,0 +1,56 @@
+"""Module for the abstract base class Equation."""
+from abc import ABC, abstractmethod
+
+
+class Equation(ABC):
+    """
+    Abstract base class for equations used in the code.
+    """
+
+    @property
+    def equation_number(self) -> str:
```

Comment:
It seems that (by design) you didn't chain the property and abstractmethod here. My question: if the property is optionally anyway: why do we define it in the ABC? We can leave it out and use it in the actuall classes we use? Now You are poluting your documentation and intellisense suggestions with options that won't work.

Reply:
You're right an equation should always have an equation_number and thus be an abstractmethod.